### PR TITLE
Add a route to authenticate user.

### PR DESCRIPTION
**Pull request by manlycode**
Adding this to give the iPad app a user ID from the rfid.

### CODE REVIEW

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Api
+  module V1
+    class UsersController < ApiController
+      def show
+        user = User.find_by(rfid: auth_params)
+        render json: Oj.dump(user, mode: :compat)
```
Comment (albus522) [inline app/controllers/api/v1/users_controller.rb]: What does this do if no user is found?

File: app/controllers/api/v1/auth_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  module V1
+    class AuthController < ApiController
+      def create
+        user = User.find_by(rfid: auth_params)
+        unless user.pours_remaining?
+          render status: :forbidden, text: "Insufficient credits remaining"
+          return
+        end
+
+        render json: Oj.dump(user, mode: :compat)
```
Comment (albus522) [inline app/controllers/api/v1/auth_controller.rb]: This is probably better as an if/else vs a return

File: app/models/user.rb [hunk 3]
```diff
@@ -33,4 +33,8 @@ def gravatar_url(opts={})
   def gravatar_base_url
     "http://www.gravatar.com/avatar/#{Digest::MD5.hexdigest((email || name).downcase.strip)}?d=retro"
   end
+
+  def pours_remaining?
+    credits.nil? || credits > 0
+  end
```
Comment (albus522) [inline app/models/user.rb]: You might want to wrap up #21 first as this will collide

**Comment by manlycode** [comment]
@albus522 Your changes look good! 👍

**Pull request merged**